src/extractor/guidance/intersection_handler.cpp

```diff
@@ -400,108 +393,105 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
 
-        out.road = &intersection[i];
-        out.same_name_id = !util::guidance::requiresNameAnnounced(
-            in_way.name_id, out_node.name_id, name_table, street_name_suffix_table);
-        if (out.same_name_id)
+        bool out_same_name_id = !util::guidance::requiresNameAnnounced(
+            in_way.name_id, out_data.name_id, name_table, street_name_suffix_table);
+        if (out_same_name_id)
             ++continue_count;
 
-        if (!out.road->entry_allowed)
-        {
-            ++continue_count_valid;
+        if (!intersection[i].entry_allowed)
             continue;
-        }
 
-        out.index = i;
-        out.deviation_from_straight = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
-        out.same_classification = out_node.road_classification == in_way.road_classification;
-        out.same_or_higher_priority =
-            out_node.road_classification.GetPriority() <= in_way.road_classification.GetPriority();
-
-        out_ways.push_back(out);
+        ++continue_count_valid;
     }
     const std::pair<std::int64_t, std::int64_t> num_continue_names =
         std::make_pair(continue_count, continue_count_valid);
 
-    const auto End = end(out_ways);
-    // dead end, or otherwise no exitable ways
-    if (out_ways.size() == 0)
-        return 0;
-
-    // sort out ways by what they share with the in way
-    const auto sort_by = [](const out_way &lhs, const out_way &rhs) {
-        auto left_tie = std::tie(
-            rhs.deviation_from_straight, lhs.same_classification, lhs.same_or_higher_priority);
-        auto right_tie = std::tie(
-            lhs.deviation_from_straight, rhs.same_classification, rhs.same_or_higher_priority);
-        return left_tie > right_tie;
+    const auto SameName = [&](const EdgeData &way_data) {
+        return way_data.name_id == in_way.name_id;
+    };
+    auto sameOrHigherPriority = [&](const auto &way_data) {
+        return way_data.road_classification.GetPriority() <=
+               in_way.road_classification.GetPriority();
+    };
+    auto IsLowPriority = [&](const auto &way_data) {
+        return way_data.road_classification.IsLowPriorityRoadClass();
+    };
+    const auto RoadCompare = [&](const ConnectedRoad &lhs, const ConnectedRoad &rhs) {
+        const EdgeData &lhs_data = node_based_graph.GetEdgeData(lhs.eid);
+        const EdgeData &rhs_data = node_based_graph.GetEdgeData(rhs.eid);
+        const auto lhs_deviation = angularDeviation(lhs.angle, STRAIGHT_ANGLE);
+        const auto rhs_deviation = angularDeviation(rhs.angle, STRAIGHT_ANGLE);
+
+        bool rhs_same_or_priority = sameOrHigherPriority(rhs_data);
+        bool rhs_low_priority = IsLowPriority(rhs_data);
+        bool lhs_same_or_priority = sameOrHigherPriority(lhs_data);
+        bool lhs_low_priority = IsLowPriority(lhs_data);
+        auto left_tie = std::tie(rhs.entry_allowed, rhs_same_or_priority, lhs_low_priority, lhs_deviation);
```

Comment:
deviation is now the least important criterion. That needs to be in the front, right? Same order as before:
`lhs.deviation_from_straight, rhs.same_classification, rhs.same_or_higher_priority`. Only keeping the `entry_allowed` in front is correct, since you filtered these before.